### PR TITLE
Cors with additional headers

### PR DESCRIPTION
**Pull request by MichalCz**
I added a small mod that allows to pass additional allowed request headers to corser.

This is done by passing: --corsheaders="Header,List,Separated,By,Commas".

### CODE REVIEW

File: bin/http-server [hunk 1]
```diff
@@ -80,6 +81,10 @@ function listen(port) {
     options.cors = true;
   }
 
+  if (argv.corsheaders) {
+    options.cors_headers = argv.corsheaders;
```
Comment (indexzero) [inline bin/http-server]: `options.corsHeaders`.

File: bin/http-server [hunk 2]
```diff
@@ -20,6 +20,7 @@ if (argv.h || argv.help) {
     "  -e --ext           Default file extension if none supplied [none]",
     "  -s --silent        Suppress log messages from output",
     "  --cors             Enable CORS via the 'Access-Control-Allow-Origin' header",
+    "  --corsheaders      Additional CORS headers",
```
Comment (indexzero) [inline bin/http-server]: This doesn't need to be a separate option. We should simply expect `--cors` to also be a string sometimes.

**Comment by indexzero** [comment]
@MichalCz will accept this once:

1. Tests are passing (mostly style it seems)
2. There are tests for this feature
3. It is implemented as a single option (e.g. `--cors` or `--cors "Header,List,Separated,By,Commas"`). No need for `--corsheaders`.

**Comment by MichalCz** [comment]
Will fix next week. :)

**Comment by MichalCz** [comment]
Hey @indexzero 

I rebased the changes, added tests, fixed changes. If you had any other requests let me know.

BTW - I still use a sepatate option in the javascript interface so that there's no need to check type in the http-server.js code, I changed command line usage only.

/Mike

**Comment by MichalCz** [comment]
Oh, wait - one more fix to do...

/Mike

**Comment by MichalCz** [comment]
@indexzero - code should now comply with what you wanted. :)

/Mike

**Comment by indexzero** [comment]
Thanks!

**Comment by danielweck** [comment]
Nice one!
Fixes https://github.com/indexzero/http-server/issues/154

**Pull request merged**